test/streams/go2go.js

```diff
@@ -0,0 +1,81 @@
+/* eslint-env mocha */
+'use strict'
+
+const chai = require('chai')
+chai.use(require('dirty-chai'))
+chai.use(require('chai-bytes'))
+const expect = chai.expect
+const { decode } = require('length-prefixed-stream')
+const { pipeline } = require('readable-stream')
+const { StreamInfo } = require('libp2p-daemon/src/protocol')
+const { ends } = require('libp2p-daemon/src/util')
+
+const spawnDaemons = require('../utils/spawnDaemons')
+
+describe('streams', () => {
+  let daemons
+
+  // Start daemons
+  before(async function () {
+    this.timeout(20 * 1000)
+
+    daemons = await spawnDaemons(2, 'go')
+
+    // connect them
+    const identify0 = await daemons[0].client.identify()
+    await daemons[1].client.connect(identify0.peerId, identify0.addrs)
+  })
+
+  // Stop daemons
+  after(async function () {
+    await Promise.all(
+      daemons.map((daemon) => daemon.stop())
+    )
+  })
+
+  it('go stream to go listener', async function () {
+    this.timeout(10 * 1000)
+
+    const sendingPeerId = await daemons[0].client.identify().peerId
+    const { receivingPeerId, receivingAddrs } = await daemons[1].client.identify()
+
+    await daemons[1].client.startServer(receivingAddrs[0], async (conn) => {
```

Comment:
you have to start the server in another address, this will be being used by the peer.  Look at https://github.com/libp2p/js-libp2p-daemon-client/blob/master/test/stream.spec.js#L68

Reply:
Switched to the addr logic from the daemon streams test